eng/Packages.Data.props

```diff
@@ -68,7 +68,7 @@
     <PackageReference Update="System.Memory.Data" Version="1.0.2" />
     <PackageReference Update="System.Numerics.Vectors" Version="4.5.0" />
     <PackageReference Update="System.Net.Http" Version="4.3.4" />
-    <PackageReference Update="System.Diagnostics.DiagnosticSource" Version="4.6.0" />
+    <PackageReference Update="System.Diagnostics.DiagnosticSource" Version="6.0.1" />
```

Comment:
How did we land on this version vs 5.x?

Reply:
5.x is deprecated, so I don't think it would be ideal to take a dependency on it. 6.0.1 is the recommended 6.x version